great_expectations/data_context/data_context.py

```diff
@@ -742,11 +743,21 @@ def datasources(self) -> Dict[str, Union[LegacyDatasource, BaseDatasource]]:
 
     @property
     def checkpoint_store_name(self):
-        return self.project_config_with_variables_substituted.checkpoint_store_name
+        try:
+            return self.project_config_with_variables_substituted.checkpoint_store_name
+        except AttributeError:
+            return DataContextConfigDefaults.DEFAULT_CHECKPOINT_STORE_NAME.value
 
     @property
     def checkpoint_store(self):
-        return self.stores[self.checkpoint_store_name]
+        checkpoint_store_name: str = self.checkpoint_store_name
+        try:
+            return self.stores[checkpoint_store_name]
+        except KeyError:
+            return self._build_store_from_config(
+                checkpoint_store_name,
+                DataContextConfigDefaults.DEFAULT_STORES.value[checkpoint_store_name],
+            )
```

Comment:
Don't we want to check if non-empty checkpoints directory exists here? Basically, call `default_checkpoints_exist`, and if not `true`, re-raise?

Reply:
Done.